### DISCUSSION TITLE
PX4 SITL AirSim heading difference

### DISCUSSION THREAD

**Discussion by istvanbeszteri** [microsoft/AirSim#3375] [Support Q&A]
I have made an Unreal AirSim PX4Multirotor drone simulation application based on Blocks. I used PX4 SITL for the drone control.
During the flight, I monitored drone orientation messages originated from PX4 SITL and found that they do not exactly match with the orientation values of the drone in Unreal Editor. Apart from X-Y coordinates are swapped there is about 8.7 degree rotation on the z axis.
Basically PX4 SITL and Unreal report slightly different orientation values.
I assume this is somehow related to magnetic vs geographic north compensations and probably dependent on GPS location, but I still do not get why the two values are different and why the value is not compensated the same way in both PX4 and AirSim.
